tests/integration/test_ingress_with_traefik_k8s.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import json
+import logging
+
+import pytest
+from helpers import initial_workload_is_ready, is_loki_up, oci_image
+
+logger = logging.getLogger(__name__)
+
+
+class AddressNotFoundError(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+@pytest.mark.abort_on_fail
+async def test_ingress_traefik_k8s(ops_test, loki_charm):
+    """Test that Loki can be related with the Grafana Agent over remote_write."""
+    loki_name = "loki"
+    traefik_name = "traefik-ingress"
+
+    await asyncio.gather(
+        ops_test.model.deploy(
+            loki_charm,
+            resources={"loki-image": oci_image("./metadata.yaml", "loki-image")},
+            application_name=loki_name,
+        ),
+        ops_test.model.deploy(
+            "traefik-k8s",
+            application_name=traefik_name,
+            channel="edge",
+            config={
+                "routing_mode": "path",
+                "external_hostname": "foo.bar",
+            },
+        ),
+    )
+
+    apps = [loki_name, traefik_name]
+    await ops_test.model.wait_for_idle(apps=apps, status="active")
+    assert initial_workload_is_ready(ops_test, apps)
```

Comment:
Asserting `initial_workload_is_ready` seems to be redundant given the previous line.

Reply:
good catch!